### PR TITLE
[tests-only][full-ci] add test to check api activities with limit and sort filter

### PR DESCRIPTION
**Pull request by prashant-gurung899**
## Description
This PR adds test to check activities with limit and sort filter (ascending | descending).
``` feature
Scenario: check activity with limit and sort filter(asc|desc)
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/9543
- Bug Report: https://github.com/owncloud/ocis/issues/10150

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by PrajwolAmatya — COMMENTED** [review]
LGTM

File: tests/acceptance/TestHelpers/GraphHelper.php [hunk 1]
```diff
@@ -2354,7 +2355,8 @@ public static function getActivities(
 		string $user,
 		string $password,
 		string $resourceId,
-		?array $filterParams = []
+		?array $filterParams = [],
+		?string $sortValue = null
```
Comment (saw-jan) [inline tests/acceptance/TestHelpers/GraphHelper.php:2359]: You should make use of `$filterParams`. it should handle all filter options
```suggestion
		?array $filterParams = [],
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 2]
```diff
@@ -2936,24 +2936,27 @@ public function thePublicTriesToCheckTheActivitiesOfResourceFromSpaceOwnedByUser
 
 	/**
 	 * @When /^user "([^"]*)" lists the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" with (depth|limit) "([^"]*)" using the Graph API/
+	 * @When /^user "([^"]*)" lists the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" with (depth|limit) "([^"]*)" and sort "([^"]*)" using the Graph API/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2939]: it deserves a separate step def

File: tests/acceptance/features/apiActivities/activities.feature [hunk 3]
```diff
@@ -2134,6 +2134,302 @@ Feature: check activities
       }
       """
 
+  @issue-10150
+  Scenario: check activity with limit and sort filter(asc|desc)
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/activities.feature:2138]:
```suggestion
  Scenario: check activities with limit and sort filters (asc/desc)
```

File: tests/acceptance/features/apiActivities/activities.feature [hunk 4]
```diff
@@ -2134,6 +2134,302 @@ Feature: check activities
...
+    Given user "Alice" has created folder "/New Folder"
+    And user "Alice" has created folder "/New Folder/Sub Folder"
+    And user "Alice" has uploaded file with content "ownCloud test text file 0" to "/New Folder/Sub Folder/textfile.txt"
+    When user "Alice" lists the activities of folder "New Folder" from space "Personal" with limit "2" and sort "desc" using the Graph API
+    Then the HTTP status code should be "200"
+    And the JSON data of the response should match
+      """
+      {
+        "type": "object",
```
Comment (saw-jan) [inline tests/acceptance/features/apiActivities/activities.feature:2147]: The main thing here is to check the order of activities. And this step doesn't do that.
So see https://github.com/owncloud/ocis/pull/10235#discussion_r1789447753

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/acceptance/TestHelpers/GraphHelper.php (hunk 1 above)
Comment (prashant-gurung899) [reply tests/acceptance/TestHelpers/GraphHelper.php:2359]: Alright. :+1:

File: tests/acceptance/bootstrap/GraphContext.php [hunk 5]
```diff
@@ -2978,6 +2978,31 @@ public function theActivitiesShouldBeInTheFollowingOrder(TableNode $table): void
 		}
 	}
 
+	/**
+	 * @When /^user "([^"]*)" lists the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" with (depth|limit) "([^"]*)" and sort "([^"]*)" using the Graph API/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2982]:
```suggestion
	 * @When /^user "([^"]*)" lists the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" with (depth|limit) "([^"]*)" and sort (asc|desc) using the Graph API/
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 6]
```diff
@@ -2978,6 +2978,31 @@ public function theActivitiesShouldBeInTheFollowingOrder(TableNode $table): void
...
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 * @param string $spaceName
+	 * @param string $filterType
+	 * @param string $filterValue
+	 * @param string $sortValue
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2989]:
```suggestion
	 * @param string $sortType
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 7]
```diff
@@ -2978,6 +2978,31 @@ public function theActivitiesShouldBeInTheFollowingOrder(TableNode $table): void
...
+	 *
+	 * @return void
+	 */
+	public function userListsTheActivitiesForFolderOfSpaceWithDepthOrLimitAndSortUsingTheGraphApi(string $user, string $resource, string $spaceName, string $filterType, string $filterValue, string $sortValue): void {
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2993]:
```suggestion
	public function userListsTheActivitiesOfResourceFromSpaceWithDepthOrLimitAndSortUsingTheGraphApi(string $user, string $resource, string $spaceName, string $filterType, string $filterValue, string $sortValue): void {
```

Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2993]: this is too long. please, break the line
```suggestion
	public function userListsTheActivitiesForFolderOfSpaceWithDepthOrLimitAndSortUsingTheGraphApi(
		string $user,
		string $resource,
		string $spaceName,
		string $filterType,
		string $filterValue,
		string $sortValue
	): void {
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 8]
```diff
@@ -2978,6 +2978,31 @@ public function theActivitiesShouldBeInTheFollowingOrder(TableNode $table): void
...
+		$resourceId = $this->featureContext->spacesContext->getResourceId($user, $spaceName, $resource);
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2994]:
```suggestion
		$resourceId = $this->spacesContext->getResourceId($user, $spaceName, $resource);
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 9]
```diff
@@ -2978,6 +2978,38 @@ public function theActivitiesShouldBeInTheFollowingOrder(TableNode $table): void
...
+	 * @When /^user "([^"]*)" lists the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" with (depth|limit) "([^"]*)" and sort "(asc|desc)" using the Graph API/
+	 *
+	 * @param string $user
+	 * @param string $resource
+	 * @param string $spaceName
+	 * @param string $filterType
+	 * @param string $filterValue
+	 * @param string $sortType
+	 *
+	 * @return void
```
Comment (anon-pradip) [inline tests/acceptance/bootstrap/GraphContext.php:2991]: I think we can add `@throws` too
```suggestion
	 * @return void
	 * @throws GuzzleException
```

File: tests/acceptance/bootstrap/GraphContext.php [hunk 10]
```diff
@@ -2978,6 +2978,39 @@ public function theActivitiesShouldBeInTheFollowingOrder(TableNode $table): void
 		}
 	}
 
+	/**
+	 * @When /^user "([^"]*)" lists the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" with (depth|limit) "([^"]*)" and sort "(asc|desc)" using the Graph API/
```
Comment (saw-jan) [inline tests/acceptance/bootstrap/GraphContext.php:2982]:
```suggestion
	 * @When /^user "([^"]*)" lists the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" with (depth|limit) "([^"]*)" and sort "(asc|desc)" using the Graph API$/
```

Comment (saw-jan) [reply tests/acceptance/bootstrap/GraphContext.php:2982]: fix other step def as well if any

**Pull request merged**